cfft/src/util.rs

```diff
@@ -0,0 +1,50 @@
+use alloc::vec::Vec;
+
+use itertools::Itertools;
+use p3_field::{ComplexExtension, Field, Powers};
+
+/// Given an integer bits, generate half of the points in the coset gH.
+/// Here H is the unique subgroup of order 2^bits and g is an element of order 2^{bits + 1}.
+/// The output will be a list {g, g^3, g^5, ...} of length size.
+/// Use size = 2^{bits} for the full domain or 2^{bits - 1} for the half domain.
+#[inline]
+pub(crate) fn cfft_domain<Base: Field, Ext: ComplexExtension<Base>>(
+    bits: usize,
+    size: usize,
+) -> Vec<Ext> {
+    let generator = Ext::circle_two_adic_generator(bits + 1);
+
+    let powers = Powers {
```

Comment:
It's better to call `.powers` or `.shifted_powers` than to create this struct directly